src/v/resource_mgmt/memory_groups.h

```diff
@@ -11,48 +11,73 @@
 
 #pragma once
 
-#include "seastarx.h"
+#include <cstddef>
 
-#include <seastar/core/memory.hh>
+/**
+ * Centralized unit for memory management.
+ *
+ * Works via a share system. First we subtract the amount of memory the user
+ * decides to use for their WebAssembly functions, then we carve out 30% for the
+ * batch cache. The remaining subsystems are distributed memory via a share
```

Comment:
The confusing part (at least for a reader of this comment) is that the batch cache will in general use _all_  the unused memory in the system. So the 30% is what we leave aside as a kind of minimum for the batch cache I guess (assuming all the subsystems stay within their limits, which in general they don't).

Reply:
Oh I guess you meant "chunk" cache not "batch" cache? They are two different things.

The way the chunk cache works is of weird. Honestly 30% is almost always just way too much memory for the chunk cache.

Reply:
I was conflating the two. I will reword this comment with your suggestions on adding the chunk cache to shares.